python/lsst/pipe/base/pipelineIR.py

```diff
@@ -531,6 +538,24 @@ def __eq__(self, other: object) -> bool:
         )
 
 
+@dataclass(eq=True)
```

Comment:
Defining equality on a raw list of dimensions is not good; it makes `[instrument, visit]` different from `[visit]`, and even worse, different from `[visit, instrument]`.  The comparison at least needs to be on sets, but I'd avoid that, too if you can, even if that means deferring some tests until `PipelineGraph` can use a `DimensionUniverse` to do it.